lib/archethic_web/api/jsonrpc/controllers/jsonrpc_controller.ex

```diff
@@ -0,0 +1,135 @@
+defmodule ArchethicWeb.API.JsonRPCController do
+  use ArchethicWeb.API, :controller
+
+  alias Archethic.TaskSupervisor
+
+  alias ArchethicWeb.API.JsonRPC.Error
+
+  alias ArchethicWeb.API.JsonRPC.Method.EstimateTransactionFee
+  alias ArchethicWeb.API.JsonRPC.Method.SendTransaction
+  alias ArchethicWeb.API.JsonRPC.Method.SimulateContractExecution
+  alias ArchethicWeb.API.JsonRPC.Method.AddOriginKey
+
+  @jsonrpc_schema :archethic
+                  |> Application.app_dir("priv/json-schemas/jsonrpc-request-2.0.json")
+                  |> File.read!()
+                  |> Jason.decode!()
+                  |> ExJsonSchema.Schema.resolve()
+
+  @methods %{
+    "send_transaction" => SendTransaction,
+    "estimate_transaction_fee" => EstimateTransactionFee,
+    "simulate_contract_execution" => SimulateContractExecution,
+    "add_origin_key" => AddOriginKey
+  }
+
+  @max_batch_size 20
+
+  def rpc(conn, param) do
+    requests = wrap_param_in_list(param)
+
+    responses =
+      if exceed_max_batch_size?(requests) do
+        create_error_response(
+          %{"jsonrpc" => "2.0", "id" => nil},
+          {:internal_error, "Batch size excedeed limit of #{@max_batch_size}"}
+        )
+        |> List.wrap()
+      else
+        execute_request_concurently(requests)
+      end
+
+    case responses do
+      [response] -> conn |> put_status(:ok) |> json(response)
+      responses -> conn |> put_status(:ok) |> json(responses)
+    end
+  end
+
+  defp wrap_param_in_list(%{"_json" => requests}) when is_list(requests), do: requests
+  defp wrap_param_in_list(%{"_json" => request}), do: List.wrap(request)
+  defp wrap_param_in_list(param), do: List.wrap(param)
+
+  defp exceed_max_batch_size?(requests), do: length(requests) > @max_batch_size
+
+  defp execute_request_concurently(requests) do
+    Task.Supervisor.async_stream(TaskSupervisor, requests, &execute_request/1,
+      on_timeout: :kill_task
+    )
+    |> Enum.zip(requests)
+    |> Enum.map(fn
+      {{:ok, {:ok, result}}, request} ->
+        create_valid_response(request, result)
+
+      {{:ok, {:error, reason}}, request} ->
+        create_error_response(request, reason)
+
+      {{:error, :timeout}, request} ->
```

Comment:
I think this should be `{:exit, :timeout}`. 

> :kill_task - the task that timed out is killed. The value emitted for that task is {:exit, :timeout}.

Reply:
Updated !
Also added Logger.warning for other exit reason